node/index.ts

```diff
@@ -8,14 +8,64 @@ import * as SignalClient from './libsignal_client';
 
 const SC = bindings('libsignal_client') as typeof SignalClient;
 
+export class PublicKey {
+  private readonly nativeHandle: SignalClient.PublicKey;
+
+  private constructor(handle: SignalClient.PublicKey) {
+    this.nativeHandle = handle;
+  }
+
+  static from_native_handle(handle: SignalClient.PublicKey): PublicKey {
+    return new PublicKey(handle);
+  }
+
+  static deserialize(buf: Buffer): PublicKey {
+    return new PublicKey(SC.PublicKey_deserialize(buf));
+  }
+
+  serialize(): Buffer {
+    return SC.PublicKey_serialize(this.nativeHandle);
+  }
+
+  verify(msg: Buffer, sig: Buffer): boolean {
+    return SC.PublicKey_verify(this.nativeHandle, msg, sig);
+  }
+
+  getNativeHandle(): SignalClient.PublicKey {
+    return this.nativeHandle;
+  }
+}
+
 export class PrivateKey {
   private readonly nativeHandle: SignalClient.PrivateKey;
 
-  constructor() {
-    this.nativeHandle = SC.PrivateKey_generate();
+  private constructor(handle: SignalClient.PrivateKey) {
+    this.nativeHandle = handle;
+  }
+
+  static generate(): PrivateKey {
+    return new PrivateKey(SC.PrivateKey_generate());
+  }
+
+  static deserialize(buf: Buffer): PrivateKey {
+    return new PrivateKey(SC.PrivateKey_deserialize(buf));
   }
 
   serialize(): Buffer {
     return SC.PrivateKey_serialize(this.nativeHandle);
   }
+
+  sign(msg: Buffer): Buffer {
+    return SC.PrivateKey_sign(this.nativeHandle, msg);
+  }
+
+  agree(other_key: PublicKey): Buffer {
+    return SC.PrivateKey_agree(this.nativeHandle, other_key.getNativeHandle());
+  }
+
+  public_key(): PublicKey {
```

Comment:
Is this more idiomatic as a no-argument method or a get-only property? (https://www.typescriptlang.org/docs/handbook/classes.html#accessors) Would probably want one of the Desktop folks to decide.

Reply:
I'd call this `getPublicKey` and keep it as a function. I rarely see accessors used in JavaScript-land. (This is anecdotal, and other Desktop devs might have different experiences.)

(Some pedantry that I hope is educational and not nitpicky: accessors are a JavaScript feature, not a TypeScript feature.)